File: hw1/ack.rb
def ack(m, n)
  if m == 0
	return n + 1
  elsif n == 0 and m > 0
	return ack(m - 1, 1)
  else
	return ack(m - 1, ack(m, n - 1))
  end
end

def main()
	puts ack(ARGV[0].to_i, ARGV[1].to_i)
end

main()

File: hw1/ack.py
import sys

def ack(m, n):
	if m == 0:
		return n + 1
	elif n == 0 and m > 0:
		return ack(m-1, 1)
	else:
		return ack(m-1, ack(m, n - 1))
def main():
	print(ack(int(sys.argv[1]), int(sys.argv[2])))

main()